cumulusci/salesforce_api/package_zip.py

```diff
@@ -13,61 +29,269 @@
+        else:
+            self._open_zip()
+        self._process()
+
+    @classmethod
+    def from_zipfile(cls, zf, *, options=None, logger=None):
+        """Start with an existing zipfile rather than a filesystem folder."""
+        return cls(zf=zf, options=options, logger=logger)
+
+    @contextlib.contextmanager
+    def _convert_sfdx_format(self, path, name):
+        orig_path = path
+        with contextlib.ExitStack() as stack:
+            if not pathlib.Path(path, "package.xml").exists():
+                self.logger.info("Converting from sfdx to mdapi format")
+                path = stack.enter_context(temporary_dir(chdir=False))
+                args = ["-r", str(orig_path), "-d", path]
+                if name:
+                    args += ["-n", name]
+                sfdx(
+                    "force:source:convert",
+                    args=args,
+                    capture_output=False,
+                    check_return=True,
+                )
+
+            yield path
+
+    def _add_files_to_package(self, path):
+        for file_path in self._find_files_to_package(path):
+            relpath = str(file_path.relative_to(path)).replace(os.sep, "/")
+            self.zf.write(file_path, arcname=relpath)
+
+    def _find_files_to_package(self, path):
+        """Generator of paths to include in the package.
+
+        Walks through all directories and files in path,
+        filtering using _include_directory and _include_file
+        """
+        for root, dirs, files in os.walk(path):
+            root_parts = pathlib.Path(root).relative_to(path).parts
+            if self._include_directory(root_parts):
+                for f in files:
+                    if self._include_file(root_parts, f):
+                        yield pathlib.Path(root, f)
+
+    def _include_directory(self, root_parts):
+        """Return boolean for whether this directory should be included in the package."""
+        # include the root directory, all non-lwc directories and sub-directories, and lwc component directories
+        return len(root_parts) == 0 or root_parts[0] != "lwc" or len(root_parts) == 2
+
+    def _include_file(self, root_parts, f):
+        """Return boolean for whether this file should be included in the package."""
+        if len(root_parts) == 2 and root_parts[0] == "lwc":
+            # is file of lwc component directory
+            lower_f = f.lower()
+            return lower_f.endswith((".js", ".js-meta.xml", ".html", ".css", ".svg"))
+        return True
+
+    def _process(self):
+        # We have to close the existing zipfile and reopen it before processing;
+        # otherwise we hit a bug in Windows where ZipInfo objects have the wrong path separators.
+        fp = self.zf.fp
+        self.zf.close()
+        self.zf = zipfile.ZipFile(fp, "r")
+
+        self._process_namespace_tokens()
+        self._clean_meta_xml()
+        self._bundle_staticresources()
+
+    def _process_namespace_tokens(self):
```

Comment:
Took me a sec to discover that this is preserving behavior from `update_dependencies`. I wonder if anyone is using this feature; I don't think I really understand the use case for tokenizing during this operation.

Reply:
It's centralizing logic that was previously done separately by both Deploy and UpdateDependencies. We clearly need namespace injection; I'm not sure why namespace_strip is a separate operation (isn't that the same as namespace injection with managed=False?); and I agree tokenizing while building a package zip doesn't make any sense. @jlantz can you provide any historic context?